central/productusage/datastore/securedunits/datastore.go

```diff
@@ -0,0 +1,36 @@
+package datastore
+
+import (
+	"context"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/stackrox/rox/central/productusage/source"
+	"github.com/stackrox/rox/generated/storage"
+)
+
+// DataStore is the datastore for usage.
+//
+//go:generate mockgen-wrapper
+type DataStore interface {
+	// Persistent storage access:
+
+	// Get returns the channel, from which the metrics could be read.
+	Get(ctx context.Context, from *types.Timestamp, to *types.Timestamp) (<-chan *storage.SecuredUnits, error)
+	// Insert puts metrics to the persistent storage.
+	Insert(ctx context.Context, metrics *storage.SecuredUnits) error
+
+	// In-memory storage access:
+
+	// AggregateAndFlush returns the aggregated metrics from the
```

Comment:
What is the time interval from which we keep these in memory?  How does that work across pod restarts?

I would also like to point out that if we ever get around to an HA central, these in memory things will complicate that effort.  IMO we should be able to accomplish all this in the database even the smoothing via averaging.  But at the same time, I do understand that were probably a fair number of design discussions that took place that I was not involved in so I'm probably missing a lot of context.

Reply:
Yeah, I guess it makes sense to call out here the thought process on why the in-memory cache.

Reply:
Secured clusters report their usage every 5 minutes. We want to store a record hourly.
The currently suggested solution: aggregate in memory, store a smoothed value.

Alternatively, using the following schema:
`Timestamp (rounded to an hour), Cluster ID, Nodes, CPU Units`
we could update records every 5 minutes / cluster, storing maximum values per cluster.

Comparing to the in-memory solution, this will multiply the storage volume by the number of clusters, and will introduce some read/write traffic to the database every 5 minutes / cluster to read and update (in the datastore, with a mutex, not under a transaction).

When querying, we can scan the DB, aggregate and smooth.

I consider the in-memory solution easier and "good enough". With the DB only solution, customers will get more details in the report, but also more DB storage consumption and central-to-db traffic.